server/src/main/kotlin/org/jetbrains/bsp/bazel/server/sync/ProjectSyncService.kt

```diff
@@ -41,12 +42,17 @@ import org.jetbrains.bsp.WorkspaceDirectoriesResult
 import org.jetbrains.bsp.WorkspaceInvalidTargetsResult
 import org.jetbrains.bsp.WorkspaceLibrariesResult
 import org.jetbrains.bsp.bazel.server.sync.model.Language
+import java.util.Optional
+import kotlin.jvm.optionals.getOrDefault
 
 /** A facade for all project sync related methods  */
 class ProjectSyncService(private val bspMapper: BspProjectMapper, private val projectProvider: ProjectProvider) {
 
-  fun initialize(cancelChecker: CancelChecker): InitializeBuildResult =
-    bspMapper.initializeServer(Language.all())
+  private var clientCapabilities = Optional.empty<BuildClientCapabilities>()
```

Comment:
You probably want to use a [lateinit var](https://kotlinlang.org/docs/properties.html#late-initialized-properties-and-variables) here